From dd1ac21a8e4eb7562d3e1a3d39437aff29674162 Mon Sep 17 00:00:00 2001
From: marshallford <inbox@marshallford.me>
Date: Wed, 27 Apr 2016 22:12:17 -0500
Subject: [PATCH] feat(webpack): allow user to serve app over local network

---
 config/index.js | 4 +++-
 package.json    | 3 ++-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/config/index.js b/config/index.js
index 79a551b..70fe44d 100644
--- a/config/index.js
+++ b/config/index.js
@@ -2,7 +2,9 @@
 import path from 'path'
 import _debug from 'debug'
 import { argv } from 'yargs'
+import ip from 'ip'
 
+const localip = ip.address()
 const debug = _debug('app:config')
 debug('Creating default configuration.')
 
@@ -24,7 +26,7 @@ const config = {
   // ----------------------------------
   // Server Configuration
   // ----------------------------------
-  server_host : 'localhost',
+  server_host : localip, // use string 'localhost' to prevent exposure on local network
   server_port : process.env.PORT || 3000,
 
   // ----------------------------------
diff --git a/package.json b/package.json
index 6cc5823..a6f8c13 100644
--- a/package.json
+++ b/package.json
@@ -83,8 +83,8 @@
     "babel-plugin-transform-runtime": "^6.3.13",
     "babel-preset-es2015": "^6.3.13",
     "babel-preset-react": "^6.3.13",
-    "babel-preset-stage-0": "^6.3.13",
     "babel-preset-react-optimize": "^1.0.1",
+    "babel-preset-stage-0": "^6.3.13",
     "babel-register": "^6.3.13",
     "babel-runtime": "^6.3.19",
     "better-npm-run": "0.0.8",
@@ -97,6 +97,7 @@
     "history": "^2.0.0",
     "html-webpack-plugin": "^2.7.1",
     "imports-loader": "^0.6.5",
+    "ip": "^1.1.2",
     "json-loader": "^0.5.4",
     "koa": "^2.0.0-alpha.3",
     "koa-connect-history-api-fallback": "^0.3.0",
